packages/core/src/utils/sql.ts

```diff
@@ -434,3 +436,56 @@ function arrayBindToNamedBind(bind: unknown[]): { [key: string]: unknown } {
 
   return out;
 }
+
+export function escapeMysqlMariaDbString(value: string): string {
```

Comment:
I'm on the fence about moving these three functions here. When we split the dialects away from core, we likely want to only ship these when using the corresponding dialects. And considering that these are not big functions, it might be fine to have them duplicated in the dialects themselves.

Reply:
Considering it's a small function i'd actually argue for the opposite and move it to the sequelize/internals package that all dialects can import

If we use the "one function per file" approach in that package, only the dialects that use it will load it, and we won't have to maintain two separate versions! :)

Reply:
That's also a good solution!